### DISCUSSION TITLE
Number of regularization images and maximum instance images

### DISCUSSION THREAD

**Discussion by dsienra** [smy20011/dreambooth-gui#92] [General]
I want to know if it's possible to increase the number of regularization images,  I have  custom images generated for the class person it have 1300 images from people from all ages and ethnicities but I think that the program is not detecting all of them, with respect to the instance images I think there is a limit of 50 images I think. 
There is a way to increase the number of regularization images and the number of instance images??
Thanks.

**Comment by futurevessel** [comment]
The images are generated automatically, so if you want more regularization images (in your case 1300), put :--num_class_images=1300 in the 'Training Arguments' box. Dreambooth-gui stores the images that it generates in order to avoid generating them over and over again, so the first time you use a particular class prompt, it will take some time generating those images, but next time you use the same class prompt, it will reuse the already generated images.

Now, if you want to use specific class images rather than having the program generate them, you need to put them where the program expects to find them, for Windows, that would be %APPDATA%/smy20011.dreambooth/, and on Linux, that would be in the .config/smy20011.dreambooth of your root account.

Place the images under the same name as the class prompt, so if for example you use 'person', then put the images in a directory called 'person' and place it in one of the the directories previously mentioned depending on which system you are using.

**Reply by dsienra** [reply]
Thanks so much for the response, do you know if there is a maximum number of instance images?
the UI only shows 50 thumbnails but maybe it still uses all of them.

**Reply by futurevessel** [reply]
I didn't know that it only showed 50 thumbs, but I'm quite certain it will use all the images in the folder for training. Dreambooth-gui is built upon https://github.com/ShivamShrirao/diffusers, and that has no limitations on the number of training images that I'm aware of, and certainly not something as low as 50 images.
